src/api-error.js

```diff
@@ -1,23 +1,27 @@
-// Custom API error to throw
-function APIError(message, data, status) {
-  let response = null;
-  let isObject = false;
+class APIError extends Error {
+  constructor(message, data, status) {
+    super(message);
+    this.name = 'APIError';
+    let response = null;
+    let isObject = false;
 
-  try {
-    response = JSON.parse(data);
-    isObject = true;
-  } catch (e) {
-    response = data;
-  }
+    try {
+      response = JSON.parse(data);
+      this.isObject = true;
+    } catch (e) {
+      response = data;
+    }
 
-  this.response = response;
-  this.message = message;
-  this.status = status;
+    this.response = response;
+    this.message = message;
+    this.status = status;
+  }
 
-  this.toString = () =>
-    `${this.message}\nResponse:\n${
-      isObject ? JSON.stringify(this.response, null, 2) : this.response
+  toString = () => {
+    return `\n${this.message}\nResponse:\n${
+      this.isObject ?  this.response : JSON.stringify(this.response, null, 2)
```

Comment:
Was this switched around accidentally?

Reply:
The switch was intentional but on second look I now realize it wasn't the best way to go about fixing an issue. I'll open a separate PR to explore the issue.

Reply:
Address in #75